flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -0,0 +1,210 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.plan.nodes.datastream
+
+import org.apache.calcite.plan.{RelOptCluster, RelTraitSet}
+import org.apache.calcite.rel.`type`.RelDataType
+import org.apache.calcite.rel.core.AggregateCall
+import org.apache.calcite.rel.{RelNode, RelWriter, SingleRel}
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.streaming.api.datastream.DataStream
+import org.apache.flink.table.api.{StreamTableEnvironment, TableException}
+import org.apache.flink.table.calcite.FlinkTypeFactory
+import org.apache.flink.table.runtime.aggregate.AggregateUtil.{CalcitePair, _}
+import org.apache.flink.table.runtime.aggregate._
+import org.apache.flink.table.plan.nodes.CommonAggregate
+import org.apache.flink.types.Row
+import org.apache.calcite.rel.core.Window
+import org.apache.calcite.rel.core.Window.Group
+import java.util.{List => JList}
+
+import org.apache.flink.table.functions.{ProcTimeType, RowTimeType}
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.IndexedSeq
+
+class DataStreamOverAggregate(
+    logicWindow: Window,
+    cluster: RelOptCluster,
+    traitSet: RelTraitSet,
+    inputNode: RelNode,
+    rowRelDataType: RelDataType,
+    inputType: RelDataType)
+  extends SingleRel(cluster, traitSet, inputNode)
+  with CommonAggregate
+  with DataStreamRel {
+
+  override def deriveRowType(): RelDataType = rowRelDataType
+
+  override def copy(traitSet: RelTraitSet, inputs: JList[RelNode]): RelNode = {
+    new DataStreamOverAggregate(
+      logicWindow,
+      cluster,
+      traitSet,
+      inputs.get(0),
+      getRowType,
+      inputType)
+  }
+
+  override def toString: String = {
+    val (
+      overWindow: Group,
+      partitionKeys: Array[Int],
+      namedAggregates: IndexedSeq[CalcitePair[AggregateCall, String]]
+      ) = genPartitionKeysAndNamedAggregates
+
+    s"Aggregate(${
+      if (!partitionKeys.isEmpty) {
+        s"partitionBy: (${groupingToString(inputType, partitionKeys)}), "
+      } else {
+        ""
+      }
+    }window: ($overWindow), " +
+      s"select: (${
+        aggregationToString(
+          inputType,
+          partitionKeys,
+          getRowType,
+          namedAggregates,
+          Seq())
+      }))"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
```

Comment:
That would be good as well. Logically, `procTime` is assigned in a Calc preceding the Window operator and has a alias assigned which is used in the order by clause of the window. So we could also show the alias instead of`PROCTIME` but both is fine with me.